plugins/modules/zos_fetch.py

```diff
@@ -606,18 +605,18 @@ def run_module():
         tmp_hlq=dict(type='qualifier_or_empty', required=False, default=None),
     )
 
-    if not module.params.get("encoding") and not module.params.get("is_binary"):
+    if not module.params.get("encoding").get("from") and not module.params.get("is_binary"):
         mvs_src = data_set.is_data_set(src)
         remote_charset = encode.Defaults.get_default_system_charset()
 
         module.params["encoding"] = {
             "from": encode.Defaults.DEFAULT_EBCDIC_MVS_CHARSET
             if mvs_src
             else remote_charset,
-            "to": module.params.get("local_charset"),
+            "to": module.params.get("encoding").get("to"),
         }
 
-    if module.params.get("encoding"):
```

Comment:
Could you add a comment here stating why we validate both .get("from") and .get("to") and not only module.params.get("encoding"). I know the reason why but is possible that in the future we won't remember and this will save us trying to reverse it .

Reply:
this is my only comment, will approve once added